src/helpers.c

```diff
@@ -782,6 +782,12 @@ int verify_fix_path(char *targetpath, struct manifest *target_MoM)
 		} else {
 			goto end;
 		}
+		/* In some circumstances (Docker using layers between updates/bundle adds,
+                 * corrupt staging content) we could have content which fails to stage.
+                 * In order to avoid this causing failure in verify_fix_path, remove the
+                 * staging content before proceeding.
+                 */
+		unlink_all_staged_content(file);
```

Comment:
`unlink_all_staged_content(file)` also unlinks the tar dotfile in /download/, so `unlink(tar_dotfile)` below can be removed.